modules/proto/src/smithytranslate/proto3/internals/Compiler.scala

```diff
@@ -619,30 +651,33 @@ private[proto3] class Compiler(model: Model, allShapes: Boolean) {
             .map(_.getNumType())
         val timestampFormatValue = extractTimestampFormat(shape)
 
-        target.accept(
-          typeVisitor(
-            isWrapped = isWrapped,
-            numType = numType,
-            timestampFormat = timestampFormatValue
+        Type.Alloy.fromShape(shape, isWrapped, isCompact).orElse {
+          target.accept(
+            typeVisitor(
+              isWrapped = isWrapped,
+              numType = numType,
+              timestampFormat = timestampFormatValue
+            )
           )
-        )
+        }
       }
 
       def operationShape(shape: OperationShape): Option[Type] = None
       def resourceShape(shape: ResourceShape): Option[Type] = None
       def serviceShape(shape: ServiceShape): Option[Type] = None
 
-      def stringShape(shape: StringShape): Option[Type] = Some {
-        val hasUUIDFormat = shape.hasTrait(classOf[alloy.UuidFormatTrait])
-        val hasProtoCompactUUID =
-          shape.hasTrait(classOf[alloy.proto.ProtoCompactUUIDTrait])
-        if (hasUUIDFormat && hasProtoCompactUUID) Type.AlloyTypes.CompactUUID
-        else if (!isWrapped) Type.String
-        else Type.GoogleWrappers.String
-      }
+      def stringShape(shape: StringShape): Option[Type] =
+        Type.Alloy.fromShape(shape, isWrapped, isCompact).orElse {
+          if (isWrapped) Some(Type.GoogleWrappers.String) else Some(Type.String)
+        }
+
       override def enumShape(shape: EnumShape): Option[Type] = {
         if (shape.hasTrait(classOf[OpenEnumTrait])) {
           Some(Type.String)
+        } else if (shape.toShapeId() == ShapeId.from("alloy#DayOfWeek")) {
+          Some(AlloyTypes.DayOfWeek)
+        } else if (shape.toShapeId() == ShapeId.from("alloy#Month")) {
+          Some(AlloyTypes.Month)
```

Comment:
Just asking purely out of curiosity:

It seems like there are a lot of functions where everything in the if-else branch is wrapped in `Some(...)`. When are these ever `None`?

FWIW, if it's useful to you, `Option.when` might be helpful somewhere.